File: arrays/distribute-candies/distribute-candies.go
package distributecandies

func distributeCandies(candyType []int) int {
	n := len(candyType)
	set := map[int]bool{}
	for _, candy := range candyType {
		set[candy] = true
	}
	if len(set) < n/2 {
		return len(set)
	}
	return n / 2
}

File: arrays/distribute-candies/distribute-candies.js
/**
 * @param {number[]} candyType
 * @return {number}
 */
function distributeCandies(candyType) {
  const n = candyType.length;
  const set = new Set();

  for (const candy of candyType) {
    set.add(candy);
  }

  return Math.min(set.size, n / 2);
}
